verde/utils.py

```diff
@@ -254,6 +255,72 @@ def grid_to_table(grid):
     return data
 
 
+def load_surfer(fname, dtype='float64'):
+    """
+    Read data from a Surfer ASCII grid file.
+
+    Surfer is a contouring, griding and surface mapping software
+    from GoldenSoftware. The names and logos for Surfer and Golden
+    Software are registered trademarks of Golden Software, Inc.
+
+    http://www.goldensoftware.com/products/surfer
+
+    This feature opens a Surfer grid file, masks any NaNs in the data,
+    and outputs a `xarray.DataArray` that contains easting and northing
+    coordinates, data values, and associated file metadata in the
+    DataArray attributes.
+
+    Parameters
+    ----------
+    fname : str
+        Name or path of the Surfer grid file
+    dtype : str or numpy dtype
+        The type of variable used for the data. Default is ``float64``. Use
+        ``float32`` if the data are large and precision is not an issue.
+    Returns
+    ----------
+    data : :class:`xarray.DataArray`
+        A 2D grid with the data variables.
+    """
+    # Surfer ASCII grid structure
+    # DSAA            Surfer ASCII GRD ID
+    # nCols nRows     number of columns and rows
+    # xMin xMax       X min max
+    # yMin yMax       Y min max
+    # zMin zMax       Z min max
+    # z11 z21 z31 ... List of Z values
+    with open(fname) as input_file:
+        # DSAA is a Surfer ASCII GRD ID (discard it for now)
+        grd_id = input_file.readline()
+        # Read the number of columns (ny) and rows (nx)
+        ydims, xdims = [int(s) for s in input_file.readline().split()]
+        # Our x points North, so the first thing we read is y, not x.
+        ymin, ymax = [float(s) for s in input_file.readline().split()]
+        xmin, xmax = [float(s) for s in input_file.readline().split()]
+        dmin, dmax = [float(s) for s in input_file.readline().split()]
+        field = np.fromiter((float(s)
+                             for line in input_file
+                             for s in line.split()),
+                            dtype=dtype)
+        nans = field >= 1.70141e+38
+        if np.any(nans):
+            field = np.ma.masked_where(nans, field)
+        err_msg = "{} of data ({}) doesn't match one from file ({})."
+        if np.allclose(dmin, field.min()) == False:
```

Comment:
```suggestion
        if not np.allclose(dmin, field.min()):
```

This should fix some of the pylint complaints. It's usually not good to compare to booleans in `if` statements.

Reply:
Rad, I wasn't positive that was the best way to do that.